vowpalwabbit/learner.h

```diff
@@ -633,6 +633,18 @@ learner<T, E> &init_learner(free_ptr<T> &dat, void (*learn)(T &, L &, E &), void
   return *ret;
 }
 
+// base learner/predictor
+// used only for python_red.cc
+template <class T, class E, class L>
+learner<T, E>& init_learner_py(T* dat, void (*learn)(T&, L&, E&), void (*predict)(T&, L&, E&), size_t params_per_weight,
```

Comment:
Please migrate this to follow the contract of the make functions. Perhaps name `make_python_base_learner` if special functionality is required